File: backend/src/main/java/com/lifestyleai/service/JournalService.java
package com.lifestyleai.service;

import java.util.List;

import com.lifestyleai.dto.journal.DailyJournalResponse;
import com.lifestyleai.dto.journal.JournalRequest;
import com.lifestyleai.dto.journal.JournalResponse;

public interface JournalService {

    JournalResponse addJournal(JournalRequest request);

    JournalResponse updateJournal(Long journalId, JournalRequest request);

    void deleteJournal(Long journalId);
    
    /**
     * Returns today's journal entries.
     */
    DailyJournalResponse getTodayJournals();

    /**
     * Returns journal history for the last N days.
     */
    List<DailyJournalResponse> getJournalHistory(
            Integer days);

}

File: frontend/src/services/journalService.ts
import { apiConnector } from "../api/apiConnector";
import { JOURNAL_API } from "../api/apiEndpoints";

import type { ApiResponse } from "../types/common";

import type {
    DailyJournalResponse,
    JournalRequest,
    JournalResponse,
} from "../types/journal";

export const getTodayJournals = async (
    userId: number,
): Promise<ApiResponse<DailyJournalResponse>> => {

    const response =
        await apiConnector<ApiResponse<DailyJournalResponse>>(
            "GET",
            `${JOURNAL_API.TODAY}/${userId}/today`,
        );

    return response.data;
};

export const getJournalHistory = async (
    userId: number,
    days: number,
): Promise<ApiResponse<DailyJournalResponse[]>> => {

    const response =
        await apiConnector<ApiResponse<DailyJournalResponse[]>>(
            "GET",
            `${JOURNAL_API.HISTORY}/${userId}/history`,
            null,
            {
                days,
            },
        );

    return response.data;
};

export const addJournal = async (
    request: JournalRequest,
): Promise<ApiResponse<JournalResponse>> => {

    const response =
        await apiConnector<ApiResponse<JournalResponse>>(
            "POST",
            JOURNAL_API.BASE,
            request,
        );

    return response.data;
};

export const updateJournal = async (
    journalId: number,
    request: JournalRequest,
): Promise<ApiResponse<JournalResponse>> => {

    const response =
        await apiConnector<ApiResponse<JournalResponse>>(
            "PUT",
            `${JOURNAL_API.BASE}/${journalId}`,
            request,
        );

    return response.data;
};

export const deleteJournal = async (
    journalId: number,
): Promise<ApiResponse<void>> => {

    const response =
        await apiConnector<ApiResponse<void>>(
            "DELETE",
            `${JOURNAL_API.BASE}/${journalId}`,
        );

    return response.data;
};